File: test/languages/sample.fs
namespace Akihabara.Fixtures

open System

type ReviewState =
    | Draft
    | Published of publishedAt: DateTimeOffset

type Review = {
    Id: Guid
    Title: string
    Score: float option
    State: ReviewState
}

module Reviews =
    let private describeScore score =
        match score with
        | Some value when value >= 8.0 -> $"recommended ({value:F1})"
        | Some value -> $"rated {value:F1}"
        | None -> "unrated"

    let summarize reviews =
        reviews
        |> Seq.filter (fun review -> match review.State with Published _ -> true | Draft -> false)
        |> Seq.map (fun review -> review.Title, describeScore review.Score)
        |> Map.ofSeq

File: test/languages/sample.php
<?php

declare(strict_types=1);

namespace Akihabara\Fixtures;

enum ReviewState: string
{
    case Draft = 'draft';
    case Published = 'published';
}

readonly class Review
{
    public function __construct(
        public int $id,
        public string $title,
        public ?float $score = null,
        public ReviewState $state = ReviewState::Draft,
    ) {}

    public function summary(): string
    {
        $score = match ($this->score) {
            null => 'unrated',
            default => number_format($this->score, 1),
        };
        return "{$this->title}: {$score}";
    }
}

$reviews = [new Review(1, 'Night Walk', 8.5, ReviewState::Published)];
echo implode(PHP_EOL, array_map(fn (Review $review): string => $review->summary(), $reviews));
